### PR TITLE
Lloggin session detailed and minor fixes

### LINKED ISSUE #5837: Cannot login: Stuck on Accept Cookies page
## Expected Behavior

Login and start working

## Current Behavior

INFO [2020-10-21 01:15:56] [my.account]  Session started!
oooooooooooooooooooooooooooooooooooooooooooooooooooooo
INFO [2020-10-21 01:15:56] [my.account]  -- Connection Checklist [1/2] (Internet Connection Status)
INFO [2020-10-21 01:15:57] [my.account]  - Internet Connection Status: ok
INFO [2020-10-21 01:15:57] [my.account]  - Current IP is "xx.yyy.zz.aa" and it's from "Germany/DE"
INFO [2020-10-21 01:15:57] [my.account]  -- Connection Checklist [2/2] (Hide Selenium Extension)
INFO [2020-10-21 01:15:57] [my.account]  - window.navigator.webdriver response: None
INFO [2020-10-21 01:15:57] [my.account]  - Hide Selenium Extension: ok
Cookie file not found, creating cookie...
INFO [2020-10-21 01:16:42] [my.account]  Timed out with failure while explicitly waiting until visibility of element located!

**********************************************************************
If raising an issue, please also upload the file located at:
/tmp/20201021-011647.html

Temp File is about accepting cookies! It appears the script is stuck on this page.

Continue to Instagram by accepting cookies?

One of the ways we use cookies is to show you useful and relevant ads on and off Instagram. You can read more about the ways we use cookies in our Cookie Policy. You can control how we use data to show you ads by using the tools described below.

Follow the instructions provided by your website or mobile browser (usually located within the "Help", "Tools" or "Edit" facility) to modify your cookie settings. Please note that if you set your browser to disable cookies or other technologies, you may not be able to access certain parts of our Service and other parts of our Service may not work properly.

To learn more about the choices that advertisers provide generally for individuals to influence how information about their online activities over time and across third-party Web sites or online services is collected and used, visit the Network Advertising Initiative, the Digital Advertising Alliance, or the European Digital Advertising Alliance.

In addition, your browser or device may offer settings that allow you to choose whether browser cookies are set and to delete them. These controls vary by browser, and manufacturers may change both the settings they make available and how they work at any time. As of 5 October 2020, you may find additional information about the controls offered by popular browsers at the links below. Certain parts of the Facebook Products may not work properly if you have disabled browser cookie use. Please be aware these controls are distinct from the controls that Facebook offers you.

## Possible Solution (optional)

Code something to ship the accept cookie page?

## InstaPy configuration

standard quickstart script

### LINKED ISSUE #5886: Stuck at "Cookie file not found, creating cookie..."
Hi guys!
Anyone could help me with an issue? Im quite sure its a very common trouble, but i cant seam to fix it :frowning:
I run the script but i get stuck in "Cookie file not found, creating cookie..." and then it crashes :frowning:

This is what i get:

`WARNING [2020-11-12 08:28:59] [USERNAME]  Cookie file not found, creating cookie...
INFO [2020-11-12 08:29:41] [USERNAME]  Timed out with failure while explicitly waiting until visibility of element located!

**********************************************************************
If raising an issue, please also upload the file located at:
C:\Users\user\AppData\Local\Temp\20201112-082946.html
**********************************************************************

INFO [2020-11-12 08:29:46] [USERNAME]  Sessional Live Report:
        |> No any statistics to show

[Session lasted 57.04 seconds]
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-12 08:29:46] [USERNAME]  Session ended!
ooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo

Traceback (most recent call last):
  File ".\instapytest.py", line 23, in <module>
    with smart_run(session):
  File "C:\Python37\lib\contextlib.py", line 114, in __enter__
    raise RuntimeError("generator didn't yield") from None
RuntimeError: generator didn't yield`

### LINKED ISSUE #5890: Couldn't follow 'Username'! ~user is inaccessible
## Expected Behavior
User is being followed
## Current Behavior
Error is being thrown: 
`Couldn't follow 'Username'! ~user is inaccessible`
It seems like selenium isn't able to get the button element. There was an attempted fix in https://github.com/timgrossmann/InstaPy/issues/5298, but that doesn't fix it for me. Maybe Instagram changed their code.
## Possible Solution (optional)
This was the presented solution in https://github.com/timgrossmann/InstaPy/issues/5298, but it's not working for me. The change was made in xpath_compile.py:
```
xpath["get_buttons_from_dialog"] = {
    "follow_button": "//div/button[text()='Follow']",
    "unfollow_button": "//div/button[text() = 'Following']",
}
```
## InstaPy configuration
Doesn't seem to be a config issue

### LINKED ISSUE #5909: Does this bot even work?
## Current Behavior
Never  been able to login to instagram with this bot. I've been getting the same error for months. Never used this successfully. 

Cookie file not found, creating cookie...
Login A/B test detected! Trying another string...

............................................................................................................................
CRITICAL [2020-11-17 22:54:39] [myusername]  Unable to login to Instagram! You will find more information in the logs above.

### LINKED ISSUE #5912: First login always successful, all login attempts after that always fail
## Expected Behavior
Logging in should always work.

## My code
```
# imports
from instapy import InstaPy
from instapy import smart_run
from instapy import set_workspace


# set workspace folder at desired location (default is at your home folder)
set_workspace(path="...")

# get an InstaPy session!
session = InstaPy(username='...', password='...')
session.login()

with smart_run(session):
    # general settings
    # session.set_dont_include(["friend1", "friend2", "friend3"])
    popeye_followers = session.grab_followers(username="...", amount="full", live_match=True, store_locally=True)
    print(popeye_followers)
```

## Current Behavior
```
Traceback (most recent call last):
  File "C:\Users\alens\Desktop\Lonya\InstaPy\quickstart.py", line 14, in <module>
    session.login()
  File "C:\Users\alens\Desktop\Lonya\InstaPy\instapy\instapy.py", line 425, in login
    if not login_user(
  File "C:\Users\alens\Desktop\Lonya\InstaPy\instapy\login_util.py", line 249, in login_user
    browser.add_cookie(cookie)
  File "C:\Users\alens\AppData\Local\Programs\Python\Python39-32\lib\site-packages\selenium-4.0.0a7-py3.9.egg\selenium\webdriver\remote\webdriver.py", line 1016, in add_cookie
    assert cookie_dict['sameSite'] in ['Strict', 'Lax']
AssertionError
```
## InstaPy configuration
Win 10 (64-bit)
Firefox 83.0 (64-bit)
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:23:07) [MSC v.1927 32 bit (Intel)] on win32

### LINKED ISSUE #5931: Sample larger than population or is negative
Hello ! First of all, I would like to thank the dev, I am discovering this tools which looks very good. But, I cannot use it for now.

```
(bots) [thibaut@neowise instapy]$ python quickstart.py 
InstaPy Version: 0.6.12
 ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._.  ._. 
Workspace in use: "/home/thibaut/InstaPy"
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-29 14:28:31] [thibautflt]  Session started!
oooooooooooooooooooooooooooooooooooooooooooooooooooooo
INFO [2020-11-29 14:28:31] [thibautflt]  -- Connection Checklist [1/2] (Internet Connection Status)
INFO [2020-11-29 14:28:31] [thibautflt]  - Internet Connection Status: ok
INFO [2020-11-29 14:28:31] [thibautflt]  - Current IP is "79.87.233.10" and it's from "France/FR"
INFO [2020-11-29 14:28:31] [thibautflt]  -- Connection Checklist [2/2] (Hide Selenium Extension)
INFO [2020-11-29 14:28:31] [thibautflt]  - window.navigator.webdriver response: None
INFO [2020-11-29 14:28:31] [thibautflt]  - Hide Selenium Extension: ok
Cookie file not found, creating cookie...
................................................................
INFO [2020-11-29 14:28:49] [thibautflt]  Logged in successfully!
''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
INFO [2020-11-29 14:28:49] [thibautflt]  Saving account progress...
______________________________________________________________________________
INFO [2020-11-29 14:29:03] [thibautflt]  Starting to follow user `Followers`..
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
INFO [2020-11-29 14:29:03] [thibautflt]  User 'les.explorateurs.auvergnats' [1/3]
INFO [2020-11-29 14:29:12] [thibautflt]  GraphQL query hash: [c76146de99bb02f6415203be841dd25a]

INFO [2020-11-29 14:29:15] [thibautflt]  Sessional Live Report:
	|> No any statistics to show

On session start was FOLLOWING 667 users & had 2976 FOLLOWERS
[Session lasted 46.36 seconds]
OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
INFO [2020-11-29 14:29:15] [thibautflt]  Session ended!
ooooooooooooooooooooooooooooooooooooooooooooooooooooooo

Traceback (most recent call last):
  File "quickstart.py", line 39, in <module>
    bot.follow_user_followers(list, amount=16, randomize=False, interact=True)
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/instapy.py", line 3415, in follow_user_followers
    person_list, simulated_list = get_given_user_followers(
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/unfollow_util.py", line 1070, in get_given_user_followers
    person_list, simulated_list = get_users_through_dialog_with_graphql(
  File "/home/thibaut/venv/bots/lib/python3.8/site-packages/instapy/unfollow_util.py", line 888, in get_users_through_dialog_with_graphql
    followers_list = random.sample(followers_list, real_amount)
  File "/usr/lib/python3.8/random.py", line 363, in sample
    raise ValueError("Sample larger than population or is negative")
ValueError: Sample larger than population or is negative
```

I have a problem with the 'follow_user_followers' function. I'm using the last version of instapy in a venv, and I modified login_util.py as in https://github.com/timgrossmann/InstaPy/issues/5886#issuecomment-727207957
I checked other issues related to the ValueError, no success. The same occurs using docker-compose.

Do you have any idea ? :o
Thank you.

### LINKED ISSUE #5944: Not a valid user: ---> Sorry, couldn't get get user bio account active
Pls help  fix this
Code:
session.login()
session.set_action_delays(enabled=True,
                           like=1,
                           comment=2,
                           follow=2,
                           unfollow=2,
                           story=5)
session.set_skip_users(skip_private=False,
                       private_percentage=100,
                       skip_no_profile_pic=False,
                       no_profile_pic_percentage=100,
                       skip_business=False,
                       skip_non_business=False,
                       business_percentage=100,
                       skip_business_categories=[],
                       dont_skip_business_categories=[],
                       skip_bio_keyword=["fenerbahçe"],
                       mandatory_bio_keywords=["pfl" , "ptfl"])
session.set_user_interact(amount=2,
                 percentage=70,
                  randomize=True,
                   media='Photo')
session.follow_likers(['_'], photos_grab_amount = 5, follow_likers_per_photo = 5, randomize=True, sleep_delay=3, interact=False)

session.end()
Error: Sorry, couldn't get get user bio account active

### PR DESCRIPTION
**Pull request by elulcao**
Hello all,

I created this PR to solve some issues with login session and other minor fixes regarding outdated xpaths,
I've been using this branch past week and everything goes well.
Thanks for considering my request, 
have a nice day.

`instapy/browser.py`: moved user_agent to general settings, now all code
uses the same agent definition

`instapy/commenters_util.py`: Added logger to `get_users_from_dialog()` to
trace actions

`instapy/instapy.py`: Added the Exception to the STDOUT when a failure

`instapy/login_util.py`: `login_user()` reordered to avoid stale elements or
timeouts due to the latest IG changes, id `SameSite` cookie. In a failure
with the cookie a new one will be created to avoid user interaction at
OS level.
In some regions a new question for saving cookies in the machine is
displayed, so it was added `accept_igcookie_dialogue` to accept the new
requirement. This check will be transparent If the page is not released in a
specific zone.

`instapy/relationship_tools.py`: Added extra prints for pretty look traces

`instapy/settings.py`: defined `user_agent` for all settings

`instapy/story_util.py`: ordered imports

`instapy/util.py: Fixed format strings when some double quotes were left.
Added a try-exept to `get_users_from_dialog` to catch stale elements

instapy/xpath_compile.py: Updated xpath for `class_selectors` and typo
in `dismiss_save_information`. Added `accept_igcookie_dialogue` for
changes in `login_util.py`

Note:

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Do not include any personal data.

Fixes # (issue)
#5834 #5886 #5890 #5909 #5912 #5931 #5944 #5837 

## How Has This Been Tested?

Verified with a real IG acct, no errors reported during execution in macOS (p3.8) and ubuntu (p3.7)

- [x] Test

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project, `black -t py34`
- [x] My changes generate no new warnings

### CODE REVIEW

File: instapy/login_util.py [hunk 1]
```diff
@@ -482,12 +507,26 @@ def login_user(
...
+        cookies_list = browser.get_cookies()
+
+        for cookie in cookies_list:
+            if "sameSite" in cookie and cookie["sameSite"] == "None":
+                cookie["sameSite"] = "Strict"
+
+        try:
+            pickle.dump(
+                cookies_list,
+                open(cookie_file, "wb"),
```
Comment (breuerfelix) [inline instapy/login_util.py]: you should close the file handler again :) or use it with the `with` statement :)

**Review by breuerfelix — CHANGES_REQUESTED** [review]

**Comment by breuerfelix** [comment]
you are a hero mate! probably already told you that but thanks for putting that much time into the project! i am 100% sure that i will speak for the whole community here :)

i also fixed some bugs again in instapy gui since you are fixing bugs over here, the gui is useable again!

**Review by timgrossmann — APPROVED** [review]
Incredible! Thanks for the updates and fixes.
As @breuerfelix already mentioned, this is huge. Especially fixing the bugs people face.

Love it!

File: instapy/login_util.py (hunk 1 above)
Comment (elulcao) [reply instapy/login_util.py]: Fixed, using `with` to auto close the handler.
Squashing commits and closing the PR.
Thanks! 

```python
        try:
            # Open the cookie file to store the data
            with open(cookie_file, "wb") as cookie_f_handler:
                pickle.dump(cookies_list, cookie_f_handler)

        except pickle.PicklingError:
            # Next time, cookie will be created for the session so we are safe
            logger.warn("- Browser cookie list could not be saved to your local...")
```

**Review by breuerfelix — APPROVED** [review]

**Comment by Sebadier** [comment]
For me the issue is still persistent, even with the latest commits.
INFO [2020-12-16 10:14:41] []  - Cookie file not found, creating cookie...
INFO [2020-12-16 10:15:20] [] Timed out with failure while explicitly waiting until visibility of element located!
[20201216-101525.html.zip](https://github.com/timgrossmann/InstaPy/files/5701584/20201216-101525.html.zip)

A solution by @xSentry that works for me is here: https://github.com/timgrossmann/InstaPy/issues/5834
Maybe somebody can merge it.

**Comment by elulcao** [comment]
@Sebadier 
That new windows is not released in the zone I'm living,
so I cannot reproduce the issue in my local, maybe the 
latest PR I sent contains a non updated xpath for that new
page. 
Hope other people send a new PR with more accurate 
information for that new window. 
thanks.

**Comment by tomnewg** [comment]
@elulcao 
Thanks for all your work on this! 
#5890 started to appear for me 4 days ago. I have never gotten this error message in the months before and I haven't changed any configuration. Any idea why this is appearing again?

Many thanks for your help

**Pull request merged**